### ISSUE TITLE
[NEW FEATURE] Unsupervised metrics for self-supervised models

### ISSUE THREAD

**Issue by Duplums** [neurospin-deepinsight/nidl#41]
**Is your feature request related to a problem? Please describe.**

While model probing on downstream tasks is useful to probe the performance of a self-supervised model, other unsupervised metrics exist (some of them being prefer or introduced for a specific model: contrastive accuracy or uniformity for simclr,  2-views or multi-views alignment for SimCLR, BYOL, MoCo, feature-wise variance or covariance matrix for Barlow Twins, VICReg). 

There is currently no official implementation of these metrics (only in specific personal repositories). I think it is useful to integrate them in nidl to monitor model's characteristics (and eventually cross-validation).  

**Describe the solution you'd like**
I would like to have these SSL metrics implemented in a uniformed way, as it is the case in sklearn or torch metrics. Maybe, only functional implementations could be performed and, in a second time, class objects could be written to properly handle distributed multi-gpu training.

**Are you planning to submit a Pull Request?**
- [X] Yes
- [ ] No
